### PR TITLE
DOCS-2567 iOS Crash Reporting and Error Tracking Edits

### PR DESCRIPTION
**Pull request by alai97**
### What and why?

Updates iOS Crash Reporting and Error Tracking page with improved information architecture and copy.

Thanks @ncreated for the insights and suggestions!

### How?

DOCS-2567

### Review checklist

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing change.

### CODE REVIEW

**Review by urseberry — APPROVED** [review]

File: docs/rum_collection/crash_reporting.md [hunk 1]
```diff
@@ -73,32 +78,70 @@ Datadog.initialize(
...
+|-----------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Yes             | dSYM files are available once [App Store Connect][6] completes processing your application's build.                                                                                                                                                                                                    |
+| No              | Xcode exports dSYM files to `$DWARF_DSYM_FOLDER_PATH` at the end of your application's build. Ensure that the `DEBUG_INFORMATION_FORMAT` build setting is set to **DWARF with dYSM File**. By default, Xcode projects only set `DEBUG_INFORMATION_FORMAT` to **DWARF with dSYM File** in release. |
```
Comment (ncreated) [inline docs/rum_collection/crash_reporting.md:94]:
```suggestion
| No              | Xcode exports dSYM files to `$DWARF_DSYM_FOLDER_PATH` at the end of your application's build. Ensure that the `DEBUG_INFORMATION_FORMAT` build setting is set to **DWARF with dYSM File**. By default, Xcode projects only set `DEBUG_INFORMATION_FORMAT` to **DWARF with dSYM File** for _Release_ configuration. |
```
Where "Release" is the name of project configuration (like below), so I believe it should be upper cased.

Screenshot 2022-03-14 at 17 49 02

File: docs/rum_collection/crash_reporting.md [hunk 2]
```diff
@@ -73,32 +78,70 @@ Datadog.initialize(
...
+
+### Upload your dYSM file
+
+dYSM files minimize an application's binary size and enable faster download speed. By uploading your dYSM file to Datadog, you gain access to the file path, line number, and code snippet of each frame in an error's related stack trace.
```
Comment (ncreated) [inline docs/rum_collection/crash_reporting.md:98]:
```suggestion
By uploading your dYSM file to Datadog, you gain access to the file path, line number, and code snippet of each frame in an error's related stack trace.
```

nit: It feels out of the context in this instruction. I think this sentence fits perfectly what we say above in "Find your dYSM file", WDYT?
```diff
### Find your dYSM file

Every iOS application produces dYSM files for each application module. 
+ These files minimize an application's binary size and enable faster download speed.
Each application version contains a set of dYSM files.
```

File: docs/rum_collection/crash_reporting.md [hunk 3]
```diff
@@ -124,24 +167,22 @@ jobs:
...
-**Note**: To configure the tool to use the EU endpoint, set the `DATADOG_SITE` environment variable to `datadoghq.eu`. To override the full URL for the intake endpoint, define the `DATADOG_DSYM_INTAKE_URL` environment variable. 
+1. Run your application on an iOS simulator or a real device. Ensure that the debugger is not attached. Otherwise, Xcode captures the crash before the iOS SDK does.
+2. Execute the code containing the crash:
 
-If your application has Bitcode enabled, download your app's dSYM files on [App Store Connect][6]. For more information, see [dSYMs commands][7].
+   ```
+   func didTapButton() {
+   fatalError(“Crash the app”)
+   }
+   ```
```
Comment (ncreated) [inline docs/rum_collection/crash_reporting.md:183]:
```suggestion
   ```swift
   func didTapButton() {
      fatalError(“Crash the app”)
   }
   ```
```

**Review by ncreated — APPROVED** [review]
Well done @alai97 ! I left few notes, let me know what you think. All this content looks correct, reads smooth and is more detailed now 💪. Thanks for help 🙂!

**Pull request merged**